### PR TITLE
Move Meeting Sidebar component to SidePanel

### PR DESCRIPTION
**Pull request by oliverguenther**
- [x] Rename profile `Sidebar` components to `SidePanel` 
- [x] Use SidePanel component internally
- [x] Use async dialog stream so that we can use `with_action` slot

https://community.openproject.org/projects/openproject/work_packages/56293/activity

### CODE REVIEW

File: modules/meeting/app/components/meetings/side_panel/attachments_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<%=
+  component_wrapper do
+    render(Primer::OpenProject::SidePanel::Section.new) do |section|
+      section.with_title { t(:label_attachment_plural) }
+      section.with_description { I18n.t('meeting.attachments.text') }
+      section.with_counter { @meeting.attachments.count }
+
+      flex_layout do |attachments_container|
+        attachments_container.with_row(mt: 2) do
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/attachments_component.html.erb]: The `mt: 2` causes unneccessary vertical spacing.
```suggestion
        attachments_container.with_row do
```

File: modules/meeting/app/components/meetings/side_panel/attachments_component.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            scheme: :link,
+            color: :default,
+            underline: false,
+            font_weight: :bold,
+            id: "meetings-add-attachments"
+          )) do |button|
+            button.with_leading_visual_icon(icon: "op-attachmentadd")
+            t("js.label_add_attachments")
+          end
+        end
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/attachments_component.html.erb]: I guess this whole block should be the `with_footer_button` slot, right? Then you can get rid of most of the system arguments, propably also the `mt: 1`.

File: modules/meeting/app/components/meetings/side_panel/details_component.html.erb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              people.with_column(mr: 2) do
+                render(Primer::Beta::Button.new(
+                  scheme: :link,
+                  underline: false,
+                  tag: :a,
+                  href: participants_dialog_meeting_path(@meeting),
+                  data: { controller: 'async-dialog' }
+                )) do
+                  t("label_meeting_show_all_participants")
+                end
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/details_component.html.erb:103]: Same as above, this should be part of the `with_footer_button` slot

File: modules/meeting/app/components/meetings/side_panel/details_dialog_component.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<%=
+  render(Primer::Alpha::Dialog.new(
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/details_dialog_component.html.erb:2]: There are some issues with the async dialog:

* When I submit a form with an error (e.g. with empty duratio), there is no error message shown. The dialog can't even be closed via the buttons any more.
* When I open the dialog, and close it via the buttons, everything is fine. But if I open the dialog again, I cannot close it any more.

File: modules/meeting/app/components/meetings/side_panel/state_component.html.erb [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    render(Primer::Beta::Button.new(
+                      scheme: :link,
+                      color: :default,
+                      underline: false,
+                      font_weight: :bold,
+                      type: :submit
+                    )) do |button|
+                      button.with_leading_visual_icon(icon: :unlock)
+                      t("label_meeting_reopen_action")
+                    end
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/state_component.html.erb]: Should also be part of the `with_footer_button` slot

File: modules/meeting/app/components/meetings/side_panel/state_component.html.erb [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                    render(Primer::Beta::Button.new(
+                      scheme: :link,
+                      color: :default,
+                      underline: false,
+                      font_weight: :bold,
+                      type: :submit
+                    )) do |button|
+                      button.with_leading_visual_icon(icon: :lock)
+                      t("label_meeting_close_action")
+                    end
```
Comment (HDinger) [inline modules/meeting/app/components/meetings/side_panel/state_component.html.erb]: Same as above

**Review by HDinger — CHANGES_REQUESTED** [review]
Hi @oliverguenther 

I left some comments. I think the biggest blocker is the dialog not working correctly.

File: modules/meeting/app/components/meetings/side_panel/details_component.html.erb (hunk 3 above)
Comment (oliverguenther) [reply modules/meeting/app/components/meetings/side_panel/details_component.html.erb:103]: Bildschirmfoto 2024-07-12 um 10 17 49

This is the mobile case and looks different

File: modules/meeting/app/components/meetings/side_panel/state_component.html.erb (hunk 5 above)
Comment (oliverguenther) [reply modules/meeting/app/components/meetings/side_panel/state_component.html.erb]: It's a form with a hidden element, so we can't use the footer button directly. I don't see how we could standardize this into the component

Comment (oliverguenther) [reply modules/meeting/app/components/meetings/side_panel/state_component.html.erb]: I found a way to inline it

**Review by HDinger — APPROVED** [review]
LGTM 👍 At least one of the failing tests is meeting related. Once that is green, it is good to merge

**Pull request merged**